data_structures/queue_using_array.cpp

```diff
@@ -1,77 +1,127 @@
-/*
-    Write a program to implement Linear Queue using array.
+/**
+ * @file
+ * @brief Implementation of Linear [Queue using array]
+ * (https://www.geeksforgeeks.org/array-implementation-of-queue-simple/).
+ * @details
+ * The Linear Queue is a data structure used for holding a sequence of
+ * values, which can be added to the end line (enqueue), removed from
+ * head of line (dequeue) and displayed.
+ * ### Algorithm
+ * Values can be added by increasing the `rear` variable by 1 (which points to 
+ * the end of the array), then assigning new value to `rear`'s element of the array.
+ * 
+ * Values can be removed by increasing the `front` variable by 1 (which points to 
+ * the first of the array), so it cannot reached any more.
+ * 
+ * @author [Pooja](https://github.com/pooja-git11)
+ * @author [Farbod Ahmadian](https://github.com/farbodahm)
+ */
+#include <iostream> /// for io operations
+#include <vector>   /// for std::vector
 
-    Functions to implement
-        Enqueue (Insertion)
-        Dequeue (Deletion)
+constexpr uint8_t max_size{10};
 
-*/
-#include <iostream>
+/**
+ * @namespace data_structures
+ * @brief Algorithms with data structures
+ */
+namespace data_structures {
 
-#define MAXSIZE 10
+/**
+ * @namespace queue_using_array
+ * @brief Functions for [Queue using Array]
+ * (https://www.geeksforgeeks.org/array-implementation-of-queue-simple/) implementation.
+ */
+namespace queue_using_array {
 
+/**
+ * @brief Queue_Array class containing the main data and also index of head and tail of the array.
+ */
 class Queue_Array {
-    int front;
-    int rear;
-    int size;
-
- public:
-    Queue_Array() {
-        front = -1;
-        rear = -1;
-        size = MAXSIZE;
-    }
-    int *arr = new int[size];
-    void enqueue(int);
+public:
+    Queue_Array();
+    void enqueue(const int8_t&);
     int dequeue();
-    void display();
+    void display() const;
```

Comment:
Add a brief description of what these functions do (see how private variables are documented).